packages/legend-query/src/stores/QueryParameterState.ts

```diff
@@ -0,0 +1,262 @@
+    );
+  }
+
+  mockValues(
+    varType: Type | undefined,
+    multiplicity: Multiplicity,
+  ): InstanceValue | undefined {
+    if ((!multiplicity.upperBound || multiplicity.upperBound > 1) && varType) {
+      const collectionInst = new CollectionInstanceValue(
+        multiplicity,
+        GenericTypeExplicitReference.create(new GenericType(varType)),
+      );
+      return collectionInst;
+    }
+    if (varType instanceof PrimitiveType) {
+      const primitiveInst = new PrimitiveInstanceValue(
+        GenericTypeExplicitReference.create(new GenericType(varType)),
+        multiplicity,
+      );
+      primitiveInst.values = [
+        createMockPrimitiveProperty(
+          varType,
+          this.parameter.name === '' ? 'myVar' : this.parameter.name,
+        ),
+      ];
+      return primitiveInst;
+    } else if (varType instanceof Enumeration) {
+      const enumValueInstance = new EnumValueInstanceValue(
+        GenericTypeExplicitReference.create(new GenericType(varType)),
+        multiplicity,
+      );
+      const mock = createMockEnumerationProperty(varType);
+      if (mock !== '') {
+        enumValueInstance.values = [
+          EnumValueExplicitReference.create(varType.getValue(mock)),
+        ];
+      }
+      return enumValueInstance;
+    }
+    return undefined;
+  }
+
+  setValues(values: InstanceValue | undefined): void {
+    this.values = values;
+  }
+
+  static createDefault(
+    queryParameterState: QueryParameterState,
+  ): ParameterState {
+    return new ParameterState(
+      queryParameterState,
+      new VariableExpression(
+        '',
+        new Multiplicity(1, 1),
+        GenericTypeExplicitReference.create(
+          new GenericType(
+            queryParameterState.queryBuilderState.graphManagerState.graph.getPrimitiveType(
+              PRIMITIVE_TYPE.STRING,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  changeVariableType(type: Type): void {
+    if (type !== this.variableType) {
+      this.parameter.genericType?.value.setRawType(type);
+      this.mockParameterValues();
+    }
+  }
+
+  changeMultiplicity(
+    lowerBound: number,
+    uppderBound: number | undefined,
+  ): void {
+    const current = this.parameter.multiplicity;
+    if (
+      current.lowerBound !== lowerBound ||
+      current.upperBound !== uppderBound
+    ) {
+      current.setLowerBound(lowerBound);
+      current.setUpperBound(uppderBound);
+      this.mockParameterValues();
+    }
+  }
+
+  get variableName(): string {
+    return this.parameter.name;
+  }
+
+  get variableType(): Type | undefined {
+    return this.parameter.genericType?.value.rawType;
+  }
+}
+
+export class QueryParameterState {
+  selectedParameter: ParameterState | undefined;
+  queryBuilderState: QueryBuilderState;
+  parameters: ParameterState[] = [];
+  isDisabled: boolean;
+  valuesEditorIsOpen = false;
+
+  constructor(
+    queryBuilderState: QueryBuilderState,
+    isDisabled?: boolean | undefined,
```

Comment:
So here's how I think we should do this so we don't end up regretting polluting the state like what we did with `EditorStore`. We should make this part of `QueryBuilderConfig` manifested in `QueryBuilderState.config`. Since it's the initial setup of `QueryBuilder` which is unlikely to change. Now, we don't need to pass this around anymore.